corehq/apps/reports_core/filters.py

```diff
@@ -178,20 +178,21 @@ class ChoiceListFilter(BaseFilter):
     Filter for a list of choices. Each choice should be a Choice object as per above.
     """
 
-    def __init__(self, name, required=True, label='Choice List Filter',
+    def __init__(self, name, datatype, required=True, label='Choice List Filter',
                  template='reports_core/filters/choice_list_filter.html',
                  css_id=None, choices=None):
         params = [
             FilterParam(name, True),
         ]
         super(ChoiceListFilter, self).__init__(required=required, name=name, params=params)
+        self.datatype = datatype
         self.label = label
         self.template = template
         self.css_id = css_id or self.name
         self.choices = choices or []
 
     def value(self, **kwargs):
-        choice = unicode(kwargs[self.name])
+        choice = transform_from_datatype(self.datatype)(kwargs[self.name]) or kwargs[self.name]
```

Comment:
I just realized this or logic doesn't work if you pass in `"0"`. made more explicit in https://github.com/dimagi/commcare-hq/pull/7273